centreon/www/include/configuration/configObject/service_categories/DB-Func.php

```diff
@@ -156,64 +167,141 @@ function multipleServiceCategorieInDB($sc = [], $nbrDup = [])
+        $serviceCategories = [$serviceCategoryId => "1"];
     }
-    foreach (array_keys($sc_arr) as $key) {
-        $query = "UPDATE service_categories SET sc_activate = '1' WHERE sc_id = :sc_id";
-        $statement = $pearDB->prepare($query);
-        $statement->bindValue(':sc_id', $key, \PDO::PARAM_INT);
-        $statement->execute();
+
+    $updateStatement = $pearDB->prepare(
+        <<<SQL
+            UPDATE service_categories
+            SET sc_activate = '1'
+            WHERE sc_id = :serviceCategoryId
+        SQL
+    );
+    $selectStatement = $pearDB->prepare(
+        <<<SQL
+            SELECT sc_name FROM `service_categories`
+            WHERE `sc_id` = :serviceCategoryId LIMIT 1
+        SQL
+    );
+    foreach (array_keys($serviceCategories) as $serviceCategoryId) {
+        $updateStatement->bindValue(':serviceCategoryId', $serviceCategoryId, \PDO::PARAM_INT);
+        $updateStatement->execute();
+
+        $selectStatement->bindValue(':serviceCategoryId', $serviceCategoryId, \PDO::PARAM_INT);
+        $selectStatement->execute();
+        $result = $selectStatement->fetch();
+        $centreon->CentreonLogAction->insertLog(
+            object_type: ActionLog::OBJECT_TYPE_SERVICECATEGORIES,
+            object_id: $serviceCategoryId,
+            object_name: $result['sc_name'],
+            action_type: ActionLog::ACTION_TYPE_ENABLE
+        );
     }
 }
 
-function disableServiceCategorieInDB(?int $sc_id = null, $sc_arr = [])
+function disableServiceCategorieInDB(?int $serviceCategoryId = null, $serviceCategories = [])
 {
-    if (!$sc_id && !count($sc_arr)) {
+    if (! $serviceCategoryId && ! count($serviceCategories)) {
         return;
     }
-    global $pearDB;
-    if ($sc_id) {
-        $sc_arr = [$sc_id => "1"];
+
+    global $pearDB, $centreon;
+    if ($serviceCategoryId) {
+        $serviceCategories = [$serviceCategoryId => "1"];
     }
-    foreach (array_keys($sc_arr) as $key) {
-        $query = "UPDATE service_categories SET sc_activate = '0' WHERE sc_id = :sc_id";
-        $statement = $pearDB->prepare($query);
-        $statement->bindValue(':sc_id', $key, \PDO::PARAM_INT);
-        $statement->execute();
+
+    $updateStatement = $pearDB->prepare(
+        <<<SQL
+            UPDATE service_categories
+            SET sc_activate = '0'
+            WHERE sc_id = :serviceCategoryId
+        SQL
+    );
+    $selectStatement = $pearDB->prepare(
+        <<<SQL
+            SELECT sc_name FROM `service_categories`
+            WHERE `sc_id` = :serviceCategoryId LIMIT 1
+        SQL
+    );
+    foreach (array_keys($serviceCategories) as $serviceCategoryId) {
+        $updateStatement->bindValue(':serviceCategoryId', $serviceCategoryId, \PDO::PARAM_INT);
+        $updateStatement->execute();
+
+        $selectStatement->bindValue(':serviceCategoryId', $serviceCategoryId, \PDO::PARAM_INT);
+        $selectStatement->execute();
+        $result = $selectStatement->fetch();
```

Comment:
You can use the new methods of CentreonDb now ;) Don't forget to catch the exception ;)

Reply:
good